numpy/lib/function_base.py

```diff
@@ -3541,7 +3541,7 @@ def _percentile(a, q, axis=None, out=None,
     elif interpolation == 'higher':
         indices = ceil(indices).astype(intp)
     elif interpolation == 'midpoint':
-        indices = floor(indices) + 0.5
+        indices = 0.5 * (floor(indices) + ceil(indices))
```

Comment:
You may also find this Wikipedia article informative as to how the percentiles are supposed to be computed: https://en.wikipedia.org/wiki/Quantile